src/lib/web-component/log-view/log-view-element.js

```diff
@@ -0,0 +1,87 @@
+class LogView extends HTMLElement {
+  constructor() {
+    super();
+
+    const shadow = this.attachShadow({ mode: 'open' });
+
+    this.logTableTemplate = document.querySelector('.log-table-template');
+    this.logTableRowTemplate = document.querySelector(
+      '.log-table-row-template'
+    );
+
+    const logTableInstance = document.importNode(
+      this.logTableTemplate.content,
+      true
+    );
+
+    this.logDetailWrapper = logTableInstance.querySelector(
+      '.log-detail-wrapper'
+    );
+    this.logDetails = logTableInstance.querySelector('#logDetails');
+    this.closeBtn = logTableInstance.querySelector('.close');
+    this.logTableWrapper = logTableInstance.querySelector('.log-table-wrapper');
+    this.tableBody = logTableInstance.querySelector('tbody');
+
+    shadow.appendChild(logTableInstance);
+  }
+
+  addNewRows(logs) {
+    for (const log of logs) {
+      const logTableRowInstance = document.importNode(
+        this.logTableRowTemplate.content,
+        true
+      );
+      logTableRowInstance.querySelector('tr')._log = log;
+
+      logTableRowInstance.querySelector('#id').textContent = log.id;
+      logTableRowInstance.querySelector('#timestamp').textContent =
+        log.timeStamp;
+      logTableRowInstance.querySelector('#apiType').textContent = log.type;
+      logTableRowInstance.querySelector('#name').textContent = log.name;
+      logTableRowInstance.querySelector('#viewType').textContent =
+        log.viewType || 'undefined';
+
+      this.tableBody.appendChild(logTableRowInstance);
+    }
+  }
+
+  openDetailSidebar(logDetails) {
+    const logString = JSON.stringify(logDetails);
+    this.logDetails.textContent = logString;
+    this.logTableWrapper.classList.add('width-60');
+    this.logDetailWrapper.removeAttribute('hidden');
+  }
+
+  closeDetailSidebar() {
+    this.logDetailWrapper.setAttribute('hidden', true);
+    this.logTableWrapper.classList.remove('width-60');
+  }
+
+  handleEvent(event) {
+    if (event.type === 'click') {
+      const logDetails = event.target.closest('tr')?._log;
+
+      if (logDetails) {
+        this.openDetailSidebar(logDetails);
+        return;
+      }
+
+      if (event.target === this.closeBtn) {
+        this.logTableWrapper.classList.remove('width-60');
```

Comment:
Remove this. This is already done in `closeDetailSidebar`.

Reply:
ahh sorry for missed it earlier :(